src/ModelManager.cpp

```diff
@@ -588,10 +615,103 @@ QVariant ModelManager::data(const QModelIndex &index, int role) const {
                     // @TODO figure out how to compile combined flag as below. 
                     // Error is "can't convert the result to QVariant."
                     // return Qt::AlignRight | Qt::AlignBaseline;
-                    return Qt::AlignRight;
+                    return Qt::AlignCenter;
+                default:
+                    return QVariant();
+            }
+    }
+
+    return QVariant();
+}
+
+RepoManager::RepoManager(QObject * parent, Settings * settings) : QAbstractTableModel(parent)
+    , settings_(settings) {}
+
+QStringList RepoManager::getRepos() {
+    QStringList urls({kModelListUrl});
+    for (auto&& nameAndUrl : settings_->externalRepos.value()) {
+        urls.append(nameAndUrl.back());
+    }
+    return urls;
+}
+
+void RepoManager::insert(QStringList new_repo) {
+    int position = settings_->externalRepos.value().size() + 1;
+    beginInsertRows(QModelIndex(),position, position);
+    settings_->externalRepos.appendToValue(new_repo);
+    endInsertRows();
+    qobject_cast<ModelManager *>(this->parent())->fetchRemoteModels(); // Fetch remote models after new a new entry was added.
```

Comment:
@jelmervdl `q_object_cast` here means that `RepoManager` needs to be aware of of the model manager. A deeper refactoring would be necessary to separate the classes cleanly. Should we put this as a todo, as there's several pressing things (native messaging, some gui revamp and pivoting)?

Reply:
You might be able to solve this by having ModelManager listen to RepoManager's `rowsInserted` signal. That is emitted because of the `beginInsertRows()` + `endInsertRows()` code above it anyway.